client/consensus/manual-seal/src/finalize_block.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Block finalization utilities
+
+use crate::rpc;
+use sp_runtime::{
+	Justification,
+	traits::Block as BlockT,
+	generic::BlockId,
+};
+use std::{
+	sync::Arc,
+	marker::PhantomData
+};
+use hash_db::Hasher;
+use client_api::backend::Backend as ClientBackend;
+
+pub struct FinalizeBlockParams<B: BlockT, H, CB> {
```

Comment:
missing docs on public items. This is a blanket rule in the Substrate and Polkadot codebases, and I hope I don't have to leave any more comments like this in the future.